adr/ADR-0003-trustyai-service-deployment-using-operator-pattern.md

```diff
@@ -0,0 +1,266 @@
+
+[^1]: Initial implementation at https://github.com/ruivieira/trustyai-service-operator
+
+### Custom Resource
+
+An example of a custom resource is:
+
+```yaml
+apiVersion: trustyai.opendatahub.io/v1alpha1
+kind: TrustyAIService
+metadata:
+  name: trustyai-service-example
+  
+spec:
+  storage:
+    format: "PVC"
+    folder: "/inputs"
+    pv: "mypv"
+    size: "1Gi"
+  data:
+    filename: "data.csv"
+    format: "CSV"
+  trustyaiMetrics:
+    schedule: "15s"
+status:
+  phase: …
+  replicas: …
+  conditions:
+  - type: Ready
+    …
+  - type: ModelMeshReady 
+    status: "True"
+    lastTransitionTime: …
+    reason: ModelMeshHealthy
+    message: ModelMesh is running and healthy.
+  - type: StorageReady
+    status: "True"
+    lastTransitionTime: …
+    reason: StorageHealthy
+    message: Storage system is functioning correctly.
+    lastUpdateTime: …
+```
+
+In this example:
+
+* `replicas` is an optional field that specifies the number of replicas of the TrustyAI service that you want to run. If not provided, the default is one replica.
+* `storage` is a mandatory field that specifies the storage details. It has two nested fields:
+  * `format` - the storage format, (example: a Persistent Volume Claim (PVC)).
+  * `folder` - the folder path where data is stored.
+  * `pv` - the name of the Persistent Volume (PV) to use (already existing).
+  * `size` - the size of the PV to use (example: 1Gi).
+* data is a mandatory field that specifies the data details. It has two nested fields:
+  * `filename` - the suffix of the file that the service uses for data.
+  * `format` - the format of the data file (example:  a CSV file).
+* `trustyaiMetrics` is a mandatory field that specifies the metrics details. It has one nested field:
+  * `schedule` - the schedule for metrics calculation, (example: every 5 seconds).
+* `status` - as part of the reconciliation process, the operator will add additional conditions, apart from the standard ones, to the custom resource to indicate the status of the deployment. These conditions will be:
+  * `ModelMeshReady`, which indicates that the ModelMesh Serving component is running.
+  * `StorageReady`, which indicates that the storage component is running.
+
+The storage, data and metrics keys consist of the only mandatory configuration fields for the TrustyAI service, at the moment. Future configuration keys can be added to the custom resource as needed.
+
+The proposed `apiVersion` and `kind` are `trustyai.opendatahub.io/v1alpha1` and `TrustyAIService`, respectively.
+
+### ModelMesh Serving Integration
+
+The operator also ensures the correct configuration of the ModelMesh Serving component. Once the TrustyAI Service is deployed and reachable, the operator will patch the ModelMesh Serving configuration to include a custom payload processor and it will be configured to point to the consumer endpoint of the deployed TrustyAI Service.
```

Comment:
I wonder if this is really OK to do.
When installing ModelMesh via operator, this patch may be rolled back, no?